app/controllers/users_controller.rb

```diff
@@ -0,0 +1,49 @@
+class UsersController < ApplicationController
+
+  before_action :require_login, except: [:new, :create]
+  before_action :validate_user, only: [:show]
+
+  def new
+    @user = User.new(user_params[:user])
+  end
+
+# trip login so user is logged in as soon as they create the account
+  def create
+    @user = User.create(user_params[:user])
+    if @user.save
+      redirect_to login_path
+      flash[:success] = "Registration successful, take your shiny new username and password for a spin"
+    else
+      flash.now[:errors] = "Registration invalid, please try again."
+      render new_user_path
+    end
+  end
+
+  def show
+    @user_id = params[:id]
+    @user = User.find(@user_id)
+  end
+
+
+  private
+
+  # this could be added to any other controllers that also require login
```

Comment:
This could also be in the application controller so you could have only one version of this method rather than duplicating the method in other controller that require login